src/libstd/sys/unix/os.rs

```diff
@@ -142,7 +142,7 @@ pub fn split_paths<'a>(unparsed: &'a OsStr) -> SplitPaths<'a> {
     let unparsed = unparsed.as_bytes();
     SplitPaths {
         iter: unparsed.split(is_colon as fn(&u8) -> bool)
```

Comment:
do we even need the `as` here? Pretty sure they should just work directly a la needless_closure